File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        minLength = float(inf)
        start = 0
        windowSum = 0
        isChanged = False
        
        for end in range(len(nums)):
            windowSum += nums[end]
            while start <= end and windowSum >= target:
                isChanged = True
                minLength = min(minLength, end-start+1)
                windowSum -= nums[start]
                start += 1
                
        return minLength if isChanged else 0
            
                        
                    

File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        if (nums.empty()) {
            return 0;
        }

        int left = 0;
        int total = 0;
        int minLen = INT_MAX;

        for (int right = 0; right < nums.size(); right++) {
            total += nums[right];
            while (total >= target) {
                minLen = min(minLen, right - left + 1);
                total -= nums[left];
                left++;
            }
        }

        return minLen == INT_MAX ? 0 : minLen;
    }
};